File: IOTConnectSDK/iotconnect-sdk/src/main/java/com/iotconnectsdk/webservices/api/IotSDKApiInterface.java
package com.iotconnectsdk.webservices.api;

import com.iotconnectsdk.webservices.requestbean.SyncServiceRequest;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Url;

public interface IotSDKApiInterface {

    @Headers("Content-Type: application/json")
    @POST
    Call<ResponseBody> sync(@Url String url, @Body SyncServiceRequest syncRequest);

    @Headers("Content-Type: application/json")
    @GET
    Call<ResponseBody> getDiscoveryApi(@Url String url);
}

File: IOTConnectSDK/iotconnect-sdk_2_1/src/main/java/com/iotconnectsdk/webservices/api/IotSDKApiInterface.kt
package com.iotconnectsdk.webservices.api

import okhttp3.ResponseBody
import retrofit2.Response
import retrofit2.http.*

interface IotSDKApiInterface {
    @Headers("Content-Type: application/json")
    @GET
    suspend fun sync(@Url url: String?): Response<ResponseBody>

    @Headers("Content-Type: application/json")
    @GET
    suspend fun getDiscoveryApi(@Url url: String?): Response<ResponseBody>
}